custom_components/sleep_as_android/__init__.py

```diff
@@ -104,7 +104,12 @@ async def unsubscribe(self):
             _LOGGER.debug("Unsubscribing")
             if self._ha_version is None:
                 await self._get_version()
-            if self._ha_version >= AwesomeVersion("2022.3.0"):
+            if self._ha_version >= AwesomeVersion("2024.6.0"):
+                self._subscription_state = subscription.async_unsubscribe_topics(
+                    hass=self.hass,
+                    new_state=self._subscription_state,
+                )
```

Comment:
It's worth noting that in 2024.6.0, `async_unsubscribe_topics` was changed to be a partial of another function. The actual keyword argument was changed to `new_state` but a stub for `async_unsubscribe_topics` was added under the old `sub_state` keyword argument, which would cause mypy and such to say this is invalid. The stub is wrong has I've submitted a PR here: https://github.com/home-assistant/core/pull/130864